syft/frameworks/torch/tensors/interpreters/chinese_remainder_theorem.py

```diff
@@ -0,0 +1,406 @@
+                result[mod] = self_[mod] + other
+
+        else:
+            assert (
+                self_.keys() == other.keys()
+            ), "Cannot add 2 CRT tensors that don't have the same moduli"
+
+            for mod in self_.keys():
+                result[mod] = self_[mod] + other[mod]
+
+        return result
+
+    __add__ = add
+    __radd__ = add
+
+    @overloaded.method
+    def sub(self, self_, other):
+
+        result = {}
+
+        if isinstance(other, int):
+            for mod in self_.keys():
+                result[mod] = self_[mod] - other
+
+        else:
+            assert (
+                self_.keys() == other.keys()
+            ), "Cannot subtract 2 CRT tensors that don't have the same moduli"
+
+            for mod in self_.keys():
+                result[mod] = self_[mod] - other[mod]
+
+        return result
+
+    __sub__ = sub
+
+    def __rsub__(self, other):
+        return (-self).sub(-other)
+
+    @overloaded.method
+    def mul(self, self_, other):
+
+        result = {}
+
+        if isinstance(other, int):
+            for mod in self_.keys():
+                result[mod] = self_[mod] * other
+
+        else:
+            assert (
+                self_.keys() == other.keys()
+            ), "Cannot multiply 2 CRT tensors that don't have the same moduli"
+
+            for mod in self_.keys():
+                result[mod] = self_[mod] * other[mod]
+
+        return result
+
+    __mul__ = mul
+    __rmul__ = mul
+
+    def div(self, other):
+        raise NotImplementedError
+
+    __truediv__ = div
+
+    def reconstruct(self):
+        """ Build the tensor in Zq with q = prod(self.child.keys())
+        satisfying the modular system represented by the tensor.
+        The algorithm consists in:
+        1) Compute N = prod(self.child.keys())
+        2) yi = N / ni where the ni's are moduli in the system
+        3) zi = yi^(-1) mod ni (we know zi exists because all the original moduli are pairwise coprime)
+        4) The result is x = sum(ai * yi * zi) where ai is the residue modulo ni in the system
+        
+        We can see that this boils down to a linear combination of ai's with coefficients yi * zi.
+        These coefficients depend only on the moduli used to represent the tensor so we don't need
+        to compute them several times for the same tensor. We can also reuse them for a tensor produced
+        by operations on self because 1) the operations can only be done between tensors with the same moduli
+        and 2) the output tensor will have the same moduli.
+        This is why we store them after having computed them once.
+        """
+        res = 0
+        N = 1
+
+        # We need to sort the dict to be sure that the coeffs are retrieved in the same order they were stored
+        for i, (mod, ai) in enumerate(sorted(self.child.items())):
+            # We need LargePrecisionTensors to reconstruct the original tensor because large values are used in the process
```

Comment:
fyi there are alternative reconstruction algorithms to avoid this issue (given that the final number is small)

Reply:
Could you point me to the method, please?